Makefile

```diff
@@ -70,7 +76,7 @@ bundle: deps
 				rm "{}"; \
 			fi' ';' ; \
 	fi && \
-	go build -tags=embedallowed -o $(OUTPUT) -trimpath -buildmode=pie -ldflags $(LDFLAGS) .
+	go build -tags=embedallowed -o $(OUTPUT) -trimpath -ldflags $(LDFLAGS) $(BUILD_MODE) .
```

Comment:
BUILD_MODE is not that explicit here, you can

either

```shell
go build [...] -buildmode=$(BUILD_MODE)
```

or

```shell
go build [...] $(EXTRA_BUILD_ARGS)
```

Reply:
-buildmode=$(BUILD_MODE) may throw error if BUILD_MODE is empty.

Reply:
I didn't understand your point. Can you explain it in more detail? @kunish

Reply:
> -buildmode=$(BUILD_MODE) may throw error if BUILD_MODE is empty.

If that's the case, we can set the default value of BUILD_MODE as `default`. See `go help buildmode`

<img width="734" alt="image" src="https://github.com/daeuniverse/dae-wing/assets/17328586/447b7f0d-4d6b-4ac3-a5f1-dd759e2caa6c">

Reply:
> I didn't understand your point. Can you explain it in more detail? @kunish

The point is if i'm a user of this makefile, the design of this `BUILD_MODE` variable does not feel very appealing and obvious to me.

Reply:
To supplement my suggestions above

```shell
make BUILD_MODE=pie
```

```shell
make EXTRA_BUILD_ARGS="-buildmode=pie"
```

Reply:
This is not designed for users. It is an internal variable in most cases unless the user wants to overwrite it.

Reply:
> This is not designed for users. It is an internal variable in most cases unless the user wants to overwrite it.

I think it's not a good design either from a developer perspective.

Reply:
See https://github.com/daeuniverse/dae-wing/pull/102/commits/616f3e560a0cdd734d7ad64578778e35829004dd